### PR TITLE
WIP: remote: split up ACI, Docker, other image handling

### PR DESCRIPTION
**Pull request by alextoombs**
Previously, all logic for fetching images was centralized around the `aciremote` package. This splits up the logic into `remote/aci`, `remote/docker`, `remote/http`, with an interface for performing the remote fetch.

Splitting up this logic allows for improved testability of each individual component, and will allow for development of better Docker image handling.

@krobertson @jpoler @mbhinder @wallyqs 

TODO:

- [ ] Fix ACI fetch integration test
- [ ] Clean up commit history
- [ ] Unit tests for ACI pulling
- [ ] Unit tests for Docker image pulling
- [ ] Unit tests for fetch package (stubbing out `Puller`)
- [ ] Integration tests?

### CODE REVIEW

**Comment by alextoombs** [comment]
Just up for preliminary reviews; lots of unit tests left to add.

File: kurmad/runner.go [hunk 1]
```diff
@@ -124,13 +124,15 @@ func (r *runner) createDirectories() error {
 // prefetchImages is used to fetch specified images on start up to pre-load
 // them.
 func (r *runner) prefetchImages() error {
-	for _, aci := range r.config.PrefetchImages {
-		_, _, err := aciremote.LoadImage(aci, true, r.imageManager)
+	for _, img := range r.config.PrefetchImages {
+		// TODO: is `r.imageManager` needed?
```
Comment (krobertson) [inline kurmad/runner.go]: yes, since that is where they're stored

File: pkg/remote/http/pull.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/apcera/kurma/pkg/remote"
+)
+
+var (
+	// Client is the http.Client that is used by RetrieveImage to download
```
Comment (krobertson) [inline pkg/remote/http/pull.go]: no function called RetrieveImage

File: pkg/remote/docker/pull.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	return nil, errors.New("only ACI-squash-pull currently supported")
+}
+
+// pullAsACI fetches a Docker image and converts it into an ACI.
+func (d *dockerPuller) pullAsACI(dockerImageURI string) (io.ReadCloser, error) {
+	if !d.convertToACI {
+		return nil, errors.New("not configured to convert to ACI")
+	}
+	dockerName := dockerImageURI[9:]
```
Comment (krobertson) [inline pkg/remote/docker/pull.go]: should this validate the length too?

**Pull request closed without merging**